doctrine/events.rst

```diff
@@ -0,0 +1,345 @@
+            // add some code to check the entity type as early as possible
+            if (!$entity instanceof Product) {
+                return;
+            }
+
+            // ... get the entity information and log it somehow
+        }
+    }
+
+The subscriber methods receive a ``LifecycleEventArgs`` object as argument,
+which gives you access to both the entity object of the event and the entity
+manager itself.
+
+If you're using the :ref:`default services.yaml configuration <service-container-services-load-example>`,
+Symfony will register the Doctrine subscriber automatically thanks to the
+:ref:`autoconfigure <services-autoconfigure>` and
+:doc:`autowiring </service_container/autowiring>` features. However, if you need
+to associate the subscriber with a specific Doctrine connection, you must define
+a service for it and :doc:`tag it </service_container/tags>` with the
+``doctrine.event_subscriber`` tag:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        services:
+            # ...
+
+            # in most applications you don't need to define a service for your
+            # subscriber (this is only needed when using a custom connection)
+            App\EventListener\DatabaseActivitySubscriber:
+                tags:
+                    - { name: 'doctrine.event_subscriber', connection: 'default' }
+
+    .. code-block:: xml
+
+        <?xml version="1.0" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:doctrine="http://symfony.com/schema/dic/doctrine">
+            <services>
+                <!-- ... -->
+
+                <!-- in most applications you don't need to define a service for your
+                     subscriber (this is only needed when using a custom connection) -->
+                <service id="App\EventListener\DatabaseActivitySubscriber">
+                    <tag name="doctrine.event_subscriber" connection="default"/>
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        use App\EventListener\DatabaseActivitySubscriber;
+
+        // in most applications you don't need to define a service for your
+        // subscriber (this is only needed when using a custom connection)
+        $container->autowire(DatabaseActivitySubscriber::class)
+            ->addTag('doctrine.event_subscriber', ['connection' => 'default'])
+        ;
+
+.. _`Doctrine`: https://www.doctrine-project.org/
+.. _`lifecycle events`: https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/events.html#lifecycle-events
+.. _`official docs about Doctrine events`: https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/events.html
+.. _`DoctrineMongoDBBundle documentation`: https://symfony.com/doc/current/bundles/DoctrineMongoDBBundle/index.html
+.. _`Entity Listeners`: https://symfony.com/doc/current/bundles/DoctrineBundle/entity-listeners.html
```

Comment:
why having some of the features documented in a `doctrine` section of the official doc, while the other is documented in the bundle doc ?